File: mycollab-web/src/main/java/com/esofthead/mycollab/schedule/email/SendingRelayEmailNotificationAction.java
package com.esofthead.mycollab.schedule.email;

import com.esofthead.mycollab.common.domain.SimpleRelayEmailNotification;

public interface SendingRelayEmailNotificationAction {
	void sendNotificationForCreateAction(
			SimpleRelayEmailNotification notification);

	void sendNotificationForUpdateAction(
			SimpleRelayEmailNotification notification);

	void sendNotificationForCommentAction(
			SimpleRelayEmailNotification notification);
}

File: mycollab-services/src/main/scala/com/esofthead/mycollab/schedule/email/SendingRelayEmailNotificationAction.scala
package com.esofthead.mycollab.schedule.email

import com.esofthead.mycollab.common.domain.SimpleRelayEmailNotification

/**
 * @author MyCollab Ltd
 * @since 5.1.0
 */
trait SendingRelayEmailNotificationAction {
    def sendNotificationForCreateAction(notification: SimpleRelayEmailNotification): Unit

    def sendNotificationForUpdateAction(notification: SimpleRelayEmailNotification): Unit

    def sendNotificationForCommentAction(notification: SimpleRelayEmailNotification): Unit
}
